### ISSUE TITLE
Doesn't work.

### ISSUE THREAD

**Issue by Vidyut** [intel/intel-extension-for-pytorch#281]
I wasted a day installing and reinstalling and compiling.

Python versions 3.8, 3.9, 3.10 (which isn't supported anyway)

`OSError: libmkl_intel_lp64.so.2: cannot open shared object file: No such file or directory`

**Comment by Vidyut** [comment]
The plugin is installed, but it can't load. Also I don't know if it is relevant that pip list uses dashes while the plugin folder and code use underscores.

`(env) vidyut@saaki:~/AI/TEST/intel-sd$ python test.py
Traceback (most recent call last):
  File "/home/vidyut/AI/TEST/intel-sd/test.py", line 1, in <module>
    import intel_extension_for_pytorch
  File "/home/vidyut/.pyenv/versions/env/lib/python3.9/site-packages/intel_extension_for_pytorch/__init__.py", line 3, in <module>
    from . import _C
ImportError: libmkl_sycl.so.2: cannot open shared object file: No such file or directory
(env) vidyut@saaki:~/AI/TEST/intel-sd$ pip list
Package                     Version
--------------------------- -------------------
intel-extension-for-pytorch 1.10.200+gpu
numpy                       1.23.4
pip                         22.3.1
setuptools                  58.1.0
torch                       1.10.0a0+git3d5f2d4
torchvision                 0.11.0+cpu
typing_extensions           4.4.0
(env) vidyut@saaki:~/AI/TEST/intel-sd$ ls /home/vidyut/.pyenv/versions/env/lib/python3.9/site-packages/
caffe2                                              numpy                   pkg_resources                torch-1.10.0a0+git3d5f2d4.dist-info  typing_extensions.py
_distutils_hack                                     numpy-1.23.4.dist-info  __pycache__                  torchvision
distutils-precedence.pth                            numpy.libs              setuptools                   torchvision-0.11.0+cpu.dist-info
intel_extension_for_pytorch                         pip                     setuptools-58.1.0.dist-info  torchvision.libs
intel_extension_for_pytorch-1.10.200+gpu.dist-info  pip-22.3.1.dist-info    torch                        typing_extensions-4.4.0.dist-info
`

**Comment by gujinghui** [comment]
This is known issue of oneMKL. Please follow this doc to preload oneMKL.
https://github.com/intel/intel-extension-for-pytorch/blob/release/xpu/1.13/docs/tutorials/performance_tuning/known_issues.md

> OSError: libmkl_intel_lp64.so.2: cannot open shared object file: No such file or directory
>
> Wrong MKL library is used when multiple MKL libraries exist in system. Preload oneMKL by:
>
>`export LD_PRELOAD=${MKL_DPCPP_ROOT}/lib/intel64/libmkl_intel_lp64.so.2:${MKL_DPCPP_ROOT}/lib/intel64/libmkl_intel_ilp64.so.2:${MKL_DPCPP_ROOT}/lib/intel64/libmkl_gnu_thread.so.2:${MKL_DPCPP_ROOT}/lib/intel64/libmkl_core.so.2:${MKL_DPCPP_ROOT}/lib/intel64/libmkl_sycl.so.2`
If you continue seeing similar issues for other shared object files, add the corresponding files under ${MKL_DPCPP_ROOT}/lib/intel64/ by LD_PRELOAD. Note that the suffix of the libraries may change (e.g. from .1 to .2), if more than one oneMKL library is installed on the system.`

BTW, our 1.13.10 release is available. Please try the new release.
https://github.com/intel/intel-extension-for-pytorch/releases/tag/v1.13.10%2Bxpu

**Comment by Vidyut** [comment]
Oh brilliant. Thank you.

**Comment by fredlarochelle** [comment]
The instructions in the [known issues](https://github.com/intel/intel-extension-for-pytorch/blob/release/xpu/1.13/docs/tutorials/performance_tuning/known_issues.md) using `export LD_PRELOAD` didn't work for me.

I ended up using `locate` to find the directories of every "problematic files" and adding them to the user `.bashrc` using `export LD_LIBRARY_PATH`. A quick reboot and it was working.

TLDR, I used the following commands: `export LD_LIBRARY_PATH=/opt/intel/oneapi/mkl/2023.0.0/lib/intel64:/opt/intel/oneapi/compiler/2023.0.0/linux/compiler/lib/intel64_lin:/opt/intel/oneapi/compiler/2023.0.0/linux/lib`, then `source /home/${user}/.bashrc`, followed by a reboot.

**Comment by bouachalazhar** [comment]
I have this :
`ImportError: libmkl_gnu_thread.so.2: cannot open shared object file: No such file or directory
`

System :

Asus ZenBook UX325EA Windows 11
WSL2 Ubuntu 22.04 Jammy
Python 3.10.6
IPEX 1.13.10+xpu
PyTorch 1.13.1
Intel i7 11th / Intel Iris Xe graphics

**Comment by jingxu10** [comment]
Please install PyTorch from Intel channel as well. The stock pytorch 1.13.1 doesn't work with ipex 1.13.10+xpu.
`python -m pip install torch==1.13.0a0 torchvision==0.14.1a0 intel_extension_for_pytorch==1.13.10+xpu -f https://developer.intel.com/ipex-whl-stable-xpu`

**Comment by bouachalazhar** [comment]
> Please install PyTorch from Intel channel as well. The stock pytorch 1.13.1 doesn't work with ipex 1.13.10+xpu. `python -m pip install torch==1.13.0a0 torchvision==0.14.1a0 intel_extension_for_pytorch==1.13.10+xpu -f https://developer.intel.com/ipex-whl-stable-xpu`

Now I have `OSError: libmkl_intel_lp64.so.2: cannot open shared object file: No such file or directory` after `import torch`
